sycl/include/CL/sycl/types.hpp

```diff
@@ -373,16 +380,45 @@ template <typename Type, int NumElements> class vec {
   }
 
 #ifdef __SYCL_USE_EXT_VECTOR_TYPE__
-  explicit vec(const DataT &arg) {
+  template <typename T = void>
+  using EnableIfNotHostHalf = typename std::enable_if<
+      !std::is_same<DataT, cl::sycl::detail::half_impl::half>::value, T>::type;
+  template <typename T = void>
+  using EnableIfHostHalf = typename std::enable_if<
+      std::is_same<DataT, cl::sycl::detail::half_impl::half>::value, T>::type;
+
+  template <typename Ty = DataT>
+  explicit vec(const EnableIfNotHostHalf<Ty> &arg) {
     m_Data = (DataType)arg;
   }
 
   template <typename Ty = DataT>
-  typename std::enable_if<std::is_fundamental<Ty>::value, vec &>::type
-  operator=(const DataT &Rhs) {
+  typename std::enable_if<
+      std::is_fundamental<Ty>::value ||
+          std::is_same<typename std::remove_const<Ty>::type, half>::value,
+      vec &>::type
```

Comment:
Yes, I need to enable this operator only if Rhs has scalar type (or half) and distinguish unoptimized version of it from optimized since optimized version is not available for host half. If you have suggestion as we can re-write it more pretty, please let me know.